fishers-exact-test/d/dfet.d

```diff
@@ -47,51 +55,40 @@ double logHypergeometricProbability(const long[] data, const ref double[] fs) {
 		fs[data[1]] -
 		fs[data[2]] -
 		fs[data[3]] -
-		fs[data[0] + data[1] + data[2] + data[3]]
-	);
+		fs[data[0] + data[1] + data[2] + data[3]];
 }
 
+// really need to do more than this pragma to make sure the optimizer doesn't cheat
+// and inline fisherExact in to main and then start using up-front knowledge of the
+// contingency table or see that they whole "loop n times" edifice doesn't change the
+// result, or something more subtle (pulling some calculations outside of the loop),
+// all of which will give misleading benchmark results.
+pragma(inline, false)
 double fisherExact(const long[] data) {
 	// sum all table values
 	const grandTotal = data.sum;
-	
+
 	// save factorial values for repeated use in the loop below
 	const factorials = logFactorial(grandTotal);
```

Comment:
is `grandTotal` actually the largest index ever visited? Could it in fact be a smaller number? Worth following all the maths through and checking I reckon.

Reply:
I put `grandTotal - 1` in instead and it crashed, so I guess that's an answer 😄